### ISSUE TITLE
page-object implementation

### ISSUE THREAD

**Issue by sesteva** [jamesottaway/js-page-object#1]
The readme illustrates a very useful and much needed pattern implementation.
I need to put together something similar, I was wondering if you had the chance to implement it? Where did you get with this? Did you find any blockers? 
Thanks

**Comment by jamesottaway** [comment]
I fully intend on implementing this library, but my knowledge of how to develop well-factored and well-tested JS is slightly lacking for the current moment.

I'm hoping to use CampJS in August as an opportunity to work on it with someone slightly more skilled than I, but that's still nearly 2 months away.

The README here was my attempt at defining the syntax up front, so I could get feedback on it before I get the chance to implement the rest of the library.

**Comment by apchamberlain** [comment]
I think you've laid the groundwork for a really excellent implementation of PageObjects here, James, better than any other I've seen in JS.

Any progress update?  If not, I plan to start a repo for an implementation of my own, giving you due credit for designing the interface, of course.

**Comment by dylang** [comment]
@apchamberlain can you provide a link when you have something to share?

**Comment by apchamberlain** [comment]
@dylang Absolutely--watch this space! :)

**Comment by sesteva** [comment]
I was actuallly done with a simple implementation on this integrated with Protractor - AngularJS testing library. 
I was getting happy and ready to share it. And then I found https://github.com/stuplum/astrolabe
It does the same thing and even added some convenient methods.
If you wanted to take the Protractor nature and replace it by simple webdriver would be fairly simple since Protractor is just an elegant wrapper around the driver to provide angular convenient methods.

**Comment by jamesottaway** [comment]
Hi all, thanks for the interest in my idea here.

@apchamberlain Your comment the other week renewed my interest in pursuing this library, but in all honesty I'm not in a position to work on it seriously for another few months. I've got a few commits on a local `spike` branch, but it's nothing worth sharing. I'd love to see what you could put together if you get time to work on an implementation.

I'm also considering some changes to the syntax in the `README.md`, which I'm pretty sure is syntactically incorrect as it stands.

**Comment by wayneseymour** [comment]
Awesomeness...I'm watching too!

**Comment by jamesottaway** [comment]
I just wanted to thank everyone for being so interested in a random idea I once had, but unfortunately never got around to implementing.

I've changed my perspective slightly on the tradeoffs of using third-party libraries, like what I hoped this one would become, so I won't be building `js-page-object` any more.

I've explained the reasoning behind this in the [deprecation notice](https://github.com/jamesottaway/js-page-object#deprecation-notice) in the `README`.

**Comment by jamesottaway** [comment]
For anyone still interested in this idea, an ex-colleague and friend of mine (@alisterscott) has a great blog post series on using WebDriverJS and Mocha, including an article on page objects.

I've added some details to the README: https://github.com/jamesottaway/js-page-object#thanks-alister

**Comment by skinnyjames** [comment]
Hey! I found this while looking for other implementations.

I'm currently working on a Page Object implementation - that can be found here:
https://github.com/skinnyjames/screen-pill

I made a proof of concept cucumber demo using the page-object library
https://github.com/skinnyjames/cucumber-js

I'll be refactoring, as well as adding supported and yet to be supported elements to the README
Any contributions and ideas for architecture are welcome!
